src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -461,6 +461,10 @@ void Connection::websocket_connected_handler(const std::string& protocol)
                     (value_2 >= get_oldest_supported_protocol_version() && value_2 <= get_current_protocol_version());
                 if (good_version) {
                     logger.detail("Negotiated protocol version: %1", value_2);
+                    // For now, grab the connection ID from the websocket if it supports it. In the future, the server
+                    // will provide the appservices connection ID via a log message.
+                    // TODO: Remove once the server starts sending the connection ID
+                    receive_appservices_request_id(m_websocket->get_appservices_request_id());
```

Comment:
`get_appservices_request_id()` didn't seem to always have a value at this point previously

Reply:
is that why we have the checks in `receive_server_log_message()`?

Reply:
The flow in `websocket_handshake_completion_handler()` will extract it from the headers before the call to `websocket_connected_handler()`

The checks in `receive_server_log_message()` is for when the server is providing the request_id and a log message is printed before it has sent this value. If it is always including the request_id in every log message, the checks aren't really needed. Also, custom platform networking implementations may not be able to provide this value, so it could be blank.